ProcessingShowPath/ProcessingShowPath.pde

```diff
@@ -1,354 +1,82 @@
+boolean drawAll = true;      //draw the paths of all participants
+boolean heatmap = true;      //draw path as a heatmap
+boolean drawArrow = true;    //draw the paths for the Arrow condition
+boolean drawNPC = true;      //draw the paths for the NPC condition
+String singleFilePath = "arrow/A12.txt";    //path of to the test file when drawing only a single test file
+
+//for drawing single path
+Path path;         
+PathDrawer drawer;  
+int drawIterator;    //used to iterate over the coordinates of the path when drawing single path
 
 void setup() {
   size(1200,900);
+  frameRate(1000000000);
+  loadPixels();  
   PImage map = loadImage("isometric.png");
   map.filter(GRAY);
   background(map);
-  frameRate(1000000000);
-  loadPixels();  
-  c = forwardColor;
   
   if(drawAll)
   {
-    drawAlpha = 2;
-    println("Files for arrow condition: ");
-    arrowFilenames = listFileNames("C:/UnityProjects/P5Game/ProcessingShowPath/arrow");
-    printArray(arrowFilenames);
-  
-    println("Files for bee condition: ");
-    beeFilenames = listFileNames("C:/UnityProjects/P5Game/ProcessingShowPath/bee");
-    printArray(beeFilenames);
-    
     if(drawArrow)
     {
+      Reader arrowReader = new Reader();
+      arrowReader.getFilesFromFolder("C:/UnityProjects/P5Game/ProcessingShowPath/arrow");
+      Path[] arrowPaths = arrowReader.readGameFiles("arrow");
+     
       println("Drawing arrow condition files");
       println("participant_number, time_going_forward, time_going_back, mistakes_made, shortcuts_taken, wander_count, wander_time_sum, finished");
-      for(int i = 0; i < arrowFilenames.length; i++)
+      for(int i = 0; i < arrowPaths.length; i++)
       {
-        reader = createReader("arrow/" + arrowFilenames[i]);
-        drawPath();
-        FloatList wanderTimes = calculateTimesAwayFromBee(returnToBeeTimes, wanderFromBeeTimes);
-        wanderTimeSum = floatListSum(wanderTimes);
-        noMovementTimeForward = floatListSum(forwardBreakTimes);
-        noMovementTimeBack = floatListSum(backBreakTimes);
-        printParticipantResult(arrowFilenames[i], wanderTimes);
-        resetVariablesForNextTestRun();
+        PathDrawer drawer = new PathDrawer(drawForward, drawBack, heatmap);  
+        drawer.drawPath(arrowPaths[i]);
+        PathAnalyser analyser = new PathAnalyser(arrowPaths[i]);
+        analyser.analysePath();
+        analyser.printResult(false);
       }
     }
     
-    if(drawBee)
+    if(drawNPC)
     {
-      println("Drawing bee condition files");
+      Reader npcReader = new Reader();
+      npcReader.getFilesFromFolder("C:/UnityProjects/P5Game/ProcessingShowPath/bee");
+      Path[] npcPaths = npcReader.readGameFiles("bee");
+      
+      println("Drawing arrow condition files");
       println("participant_number, time_going_forward, time_going_back, mistakes_made, shortcuts_taken, wander_count, wander_time_sum, finished");
```

Comment:
these 2 parts for drawing Arrow and NPC are almost literally the same, so you could make it as a single function. Just pass some arguments, like `conditionName`, which would be `"arrow"` or  `"bee"` and kind of that's it?

Reply:
Also make that function with an argument `shouldDraw`, where you would pass `drawNPC` or `drawArrow`

Reply:
what if we want to draw both?

Reply:
those 2 functions could be wrapped around that `if`. So you'd have something like:

if (drawAll)
{
    drawCondition("arrow", drawArrow);
    drawCondition("bee", drawNPC);
}